lexers/mediawiki.lua

```diff
@@ -1,74 +1,68 @@
 -- Copyright 2006-2025 Mitchell. See LICENSE.
 -- MediaWiki LPeg lexer.
 -- Contributed by Alexander Misel.
+
 local lexer = lexer
-local P, S, B = lpeg.P, lpeg.S, lpeg.B
+local P, S = lpeg.P, lpeg.S
+
 local lex = lexer.new(...)
 
+-- Comments (high priority to avoid conflicts)
+lex:add_rule('comment', lex:tag(lexer.COMMENT, lexer.range('<!--', '-->')))
+
 -- HTML-like tags
 local dq_str = P('"') * ((lexer.any - S('>"\\')) + ('\\' * lexer.any))^0 * P('"')
 local unquoted_attr = (lexer.any - (S('"' .. "'" .. '<>=') + lexer.space))^1
+local tag_name = lexer.alpha^1
 local tag_attr = lex:tag(lexer.ATTRIBUTE, lexer.alpha^1 * lexer.space^0 *
-	('=' * lexer.space^0 * (dq_str + unquoted_attr))^-1 * lexer.space^0)
-local tag_name = lexer.alnum^1
-local tag_inner_content = lexer.space^0 * tag_attr^0 * lexer.space^0 * P('/')^-1
-
--- The tag rule should consume from '<' to the final '>'
-lex:add_rule('tag', lex:tag(lexer.TAG, '<' * P('/')^-1 * tag_name * tag_inner_content * P('>')))
+	 ('=' * lexer.space^0 * (dq_str + unquoted_attr))^-1 * lexer.space^0)
+lex:add_rule('tag', lex:tag(lexer.TAG, '<' * P('/')^-1 * tag_name *
+	(lexer.space^1 * tag_attr)^0 * lexer.space^0 * P('/')^-1 * '>'))
 
--- Internal Link: [[Target]] or [[Target|Display Text]]
--- The content can contain almost anything except unbalanced square brackets.
--- We'll highlight the whole thing as LINK.
-local internal_link_content = (lexer.any - P(']]'))^1 -- Matches everything until ']]'
-lex:add_rule('internal_link', lex:tag(lexer.LINK, P('[[') * internal_link_content * P(']]')))
+-- Internal Links
+lex:add_rule('internal_link', lex:tag(lexer.LINK, lexer.range('[[', ']]')))
 
--- External Link: [http://example.com Link text] or [http://example.com]
--- Content should start with a protocol (http/s, ftp, mailto etc.)
-local protocol = lexer.alpha^2 * P('://')
-local external_link_content = (protocol * (lexer.any - P(']'))^1) + (lexer.any - P(']'))^1
-lex:add_rule('external_link', lex:tag(lexer.LINK, P('[') * external_link_content * P(']')))
+-- External Links
+lex:add_rule('external_link', lex:tag(lexer.LINK,
+	P('[') * lexer.word_match(lexer.TYPE) * P('://') *
```

Comment:
I think you meant to use `lex:word_match(lexer.TYPE)`. That probably explains the error about the lexer not having a word list.

Reply:
Oh! Thank you! That’s it!